src/test/java/com/kodilla/ecommercee/domain/ProductTestSuite.java

```diff
@@ -0,0 +1,165 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.repository.GroupRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ProductTestSuite {
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Transactional
+    @Test
+    public void testSaveProduct() {
+        //given
+        Product testProduct = new Product();
```

Comment:
wszędzie mamy all-args-constructor użyty a w tym teście leci bezparametrowy. To celowo? Wydaje mi się, że można ujednolicić.. no chyba, że testujemy setery? :-P

Reply:
Dokładnie, po "upublicznieniu" setterów chciałem w jednym teście je przetestować, tak dla pewności, że działają. Po sprawdzeniu, w pozostałych testach używałem all-argsów ponieważ były wygodniejsze niż za każdym razem korzystać z setterów, zwłąszcza przy kilku obiektach :)

Reply:
w takim wypadku ten test testuje 2 rzeczy a jedną niejawnie... a powinien być unit ;) kwestia sporna czy należy testować settery... jeśli to jest typowy setter bez dodatkowej logiki - IMHO nie powinno się tego testować, a jeśli chcemy to testować to warto osobny test pod to zrobić i zrobić resztę testów jako zależnych. W sensie, że jak konstruktor nie działa poprawnie czy setter - to po co testować logikę zapisu danych, prawda? 

Na chwilę obecną OK :)